frontend/src/metabase/entities/actions/actions.ts

```diff
@@ -124,28 +125,25 @@ const Actions = createEntity({
     "archived",
   ],
   objectActions: {
-    createPublicLink: createAction(
+    createPublicLink: createThunkAction(
       CREATE_PUBLIC_LINK,
-      ({ id }: { id: WritebackActionId }) => {
-        return ActionsApi.createPublicLink({ id }).then(
-          ({ uuid }: { uuid: string }) => {
-            return {
-              id,
-              uuid,
-            };
-          },
-        );
-      },
+      ({ id }: { id: WritebackActionId }) =>
+        async (dispatch: Dispatch) => {
+          const { data } = await (dispatch(
+            createActionPublicLink.initiate({ id }),
+          ) as Promise<{ data: { uuid: string } }>);
+
+          return { id, uuid: data.uuid };
+        },
     ),
```

Comment:
instead of updating the action entity, can we just use RTK hooks directly in the UI components?

Reply:
I'm thinking I can set that up in another PR - I didn't want to make the PR too wild here (ironic, I know, considering the other PR)